### PR TITLE
[auth:list] Auth list filtering by hostname.

### LINKED ISSUE #21: allow auth list to filter by --apihost
`nim auth list` lists all identities across all hosts, and I'm suggesting here to allow filtering by `--apihost` to only show identities on that host.

### PR DESCRIPTION
**Pull request by rabbah**
Closes #21.

### CODE REVIEW

**Comment by rabbah** [comment]
Simpler to review the changes ignoring white space https://github.com/nimbella/nimbella-cli/pull/72/files?diff=unified&w=1.

File: src/commands/auth/list.ts [hunk 1]
```diff
@@ -25,24 +26,35 @@ const MAYBE = '   -?-  '
...
-        logger.log('')
+    if (flags.apihost) {
+      const host = parseAPIHost(flags.apihost)
+      if (host in dict) {
         logger.log(`${bold(`On API host ${host}:`)}`)
         await this.formatCredentialList(dict[host], logger)
       }
+    } else {
+      if (Object.keys(dict).length === 1) {
+        await this.formatCredentialList(Object.values(dict)[0], logger)
```
Comment (rabbah) [inline src/commands/auth/list.ts:46]: I wasn't sure why this is here as a special case - but I admit to not trying to understand. I didn't really modify the existing logic when `--apihost` is omitted. There is a small change in formatting though: the leading line that's printed when `nim auth list` is run is not removed.

File: src/commands/auth/list.ts [hunk 2]
```diff
@@ -25,24 +26,35 @@ const MAYBE = '   -?-  '
...
-    if (Object.keys(dict).length === 1) {
-      await this.formatCredentialList(Object.values(dict)[0], logger)
-    } else {
-      for (const host in dict) {
-        const rows = dict[host]
-        if (rows.length === 0) continue
-        logger.log('')
+    if (flags.apihost) {
+      const host = parseAPIHost(flags.apihost)
+      if (host in dict) {
```
Comment (rabbah) [inline src/commands/auth/list.ts:40]: no output is produced if there are no entries.

File: src/commands/auth/list.ts [hunk 3]
```diff
@@ -25,24 +26,35 @@ const MAYBE = '   -?-  '
 export default class AuthList extends NimBaseCommand {
   static description = 'List all your Nimbella namespaces'
 
-  static flags: typeof NimBaseCommand.flags = {
+  static flags = {
+    apihost: flags.string({ description: 'Only list namespaces for the specififed API host' }),
```
Comment (joshuaauerbachwatson) [inline src/commands/auth/list.ts:30]:
```suggestion
    apihost: flags.string({ description: 'Only list namespaces for the specified API host' }),
```

**Review by joshuaauerbachwatson — APPROVED** [review]
lgtm.  I corrected one typo in help text.

**Pull request merged**